entity-service/src/test/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationServiceTest.kt

```diff
@@ -0,0 +1,176 @@
+        assertUserHasRightOnEntity("rCanAdmin", neo4jAuthorizationService::userHasAdminRightsOnEntity)
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = ["rCanRead", "rCanWrite", ""])
+    fun `it should find user has not admin right on entity`(right: String) {
+        assertUserHasNotRightOnEntity(right, neo4jAuthorizationService::userHasAdminRightsOnEntity)
+    }
+
+    @Test
+    fun `it should find admin user has admin, read or write right entity`() {
+        every { neo4jAuthorizationRepository.getUserRoles("urn:ngsi-ld:User:mock-user") } returns listOf("admin")
+        every {
+            neo4jAuthorizationRepository.getAvailableRightsForEntities(
+                "urn:ngsi-ld:User:mock-user",
+                listOf("entityId")
+            )
+        } returns listOf(
+            Neo4jAuthorizationRepository.AvailableRightsForEntity(
+                targetEntityId = "entityId"
+            )
+        )
+
+        assert(neo4jAuthorizationService.userHasAdminRightsOnEntity("entityId", "mock-user"))
+        assert(neo4jAuthorizationService.userHasReadRightsOnEntity("entityId", "mock-user"))
+        assert(neo4jAuthorizationService.userHasWriteRightsOnEntity("entityId", "mock-user"))
+    }
+
+    @Test
+    fun `it shoud filter entities which user has read right`() {
+        val entitiesId = listOf("entityId", "entityId2", "entityId3", "entityId4", "entityId5")
+
+        every { neo4jAuthorizationRepository.getUserRoles("urn:ngsi-ld:User:mock-user") } returns listOf()
+        every {
+            neo4jAuthorizationRepository.getAvailableRightsForEntities("urn:ngsi-ld:User:mock-user", entitiesId)
+        } returns listOf(
+            Neo4jAuthorizationRepository.AvailableRightsForEntity(
+                targetEntityId = "entityId",
+                right = Relationship(type = listOf(AUTHORIZATION_ONTOLOGY + "rCanRead"))
+            ),
+            Neo4jAuthorizationRepository.AvailableRightsForEntity(targetEntityId = "entityId2"),
+            Neo4jAuthorizationRepository.AvailableRightsForEntity(
+                targetEntityId = "entityId3",
+                grpRight = Relationship(type = listOf(AUTHORIZATION_ONTOLOGY + "rCanAdmin"))
+            ),
+            Neo4jAuthorizationRepository.AvailableRightsForEntity(
+                targetEntityId = "entityId4",
+                right = Relationship(type = listOf(AUTHORIZATION_ONTOLOGY + "rCanRead")),
+                grpRight = Relationship(type = listOf(AUTHORIZATION_ONTOLOGY + "rCanWrite"))
+            ),
+            Neo4jAuthorizationRepository.AvailableRightsForEntity(targetEntityId = "entityId5")
+        )
+
+        assert(
+            neo4jAuthorizationService.filterEntitiesUserHasReadRight(entitiesId, "mock-user") == listOf(
+                "entityId",
+                "entityId3",
+                "entityId4"
+            )
+        )
+    }
+
+    @Test
+    fun `it shoud filter entities if user's group has read right`() {
```

Comment:
I don't get this test ... you meant `grpRight` instead of `right` at line 126?

Reply:
Indeed, and it doesn't make sense now that I'll follow your advice to merge in a single variable.